stdlib/public/core/Codable.swift

```diff
@@ -5617,14 +5617,16 @@ extension Dictionary: Encodable where Key: Encodable, Value: Encodable {
       // Since the keys are already Strings, we can use them as keys directly.
       var container = encoder.container(keyedBy: _DictionaryCodingKey.self)
       for (key, value) in self {
-        let codingKey = _DictionaryCodingKey(stringValue: key as! String)
+        let codingKey = _DictionaryCodingKey(
+          stringValue: _identityCast(key, to: String.self))
```

Comment:
Couldn't this be `_specialize` as well?

Reply:
It could! However, in this case we know that `Key` is supposed to be a `String`, so using `_identityCast` avoids having to force-unwrap its optional result. (I expect generated code to look the same, but this avoids the noise of the extra `!` or `_unsafelyUnwrapped` call.) 🤔